### PR TITLE
rfb: add RFB frames 

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://suricata.readthedocs.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/5717
 
Previous PR: #8313 

Describe changes:
- Fix `frame_len`

suricata-verify-pr: 1048

### CODE REVIEW

**Comment by catenacyber** [comment]
Are you able to investigate the CI failure ?

**Comment by hsadia538** [comment]
> Are you able to investigate the CI failure ?

Yes I think so. It's because of the suricata verify pr issue. I have made a new pr for suricata verify tests. Is there a way to trigger the CI checks again without making a new PR?

**Comment by inashivb** [comment]
> > Are you able to investigate the CI failure ?
> 
> Yes I think so. It's because of the suricata verify pr issue. I have made a new pr for suricata verify tests. Is there a way to trigger the CI checks again without making a new PR?

You need special permissions for that. For now, I have triggered the checks again. Let's see if the right branch gets picked up now.

**Review by inashivb — CHANGES_REQUESTED** [review]
Nice work, Haleema! :)
Some comments from me inline.

File: rust/src/rfb/rfb.rs [hunk 1]
```diff
@@ -151,8 +157,16 @@ impl RFBState {
...
 
-    fn parse_request(&mut self, input: &[u8]) -> AppLayerResult {
+    fn parse_request(&mut self, flow: *const Flow, stream_slice: StreamSlice) -> AppLayerResult {
         // We're not interested in empty requests.
+        let input = stream_slice.as_slice();
+        let _pdu = Frame::new(
+            flow,
+            &stream_slice,
+            input,
+            -1_i64,
```
[the remark this answers is not in the document]
Comment (hsadia538) [reply rust/src/rfb/rfb.rs:167]: From your comments, what I understand is that we don't know the length of the pdu at this point so `-1_i64` won't work either.
Should I skip the length here altogether?

File: rust/src/rfb/rfb.rs [hunk 2]
```diff
@@ -280,7 +294,15 @@ impl RFBState {
...
     }
 
-    fn parse_response(&mut self, input: &[u8]) -> AppLayerResult {
+    fn parse_response(&mut self, flow: *const Flow, stream_slice: StreamSlice) -> AppLayerResult {
+        let input = stream_slice.as_slice();
+        let _pdu = Frame::new(
+            flow,
+            &stream_slice,
+            input,
+            -1_i64,
```
[the remark this answers is not in the document]
Comment (hsadia538) [reply rust/src/rfb/rfb.rs:312]: Should we have a checker here and then create frames based on the check?

File: rust/src/rfb/rfb.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (inashivb) [reply rust/src/rfb/rfb.rs:167]: > Should I skip the length here altogether?

Don't know how can we do that.

But, apologies for being unclear, I did not mean that `-1` won't work. I meant that this is fine but should not be final thing. The length of the frame should be updated when we know about it. e.g. in telnet we do this once we know the length: https://github.com/OISF/suricata/blob/master/rust/src/telnet/telnet.rs#L306

[the remark this answers is not in the document]
Comment (hsadia538) [reply rust/src/rfb/rfb.rs:167]: Ok, so as I understand, I have to update the length of the frame. In the `rfb.rs` we are matching different `RFBGlobalState` so the `set_len` will go in each match statement within the `parse_request` and `parse_response` I have to use `set_len` in each matching block.

**Comment by hsadia538** [comment]
Followed by #8337

**Pull request closed without merging**